forte/processors/data_augment/algorithms/skeleton_op.py

```diff
@@ -0,0 +1,765 @@
+        self._entry_maps[pid] = entry_map
+        return new_pack
+
+    def _overlap_with_existing(self, pid: int, begin: int, end: int) -> bool:
+        r"""
+        This function will check whether the new span
+        has an overlap with any existing spans.
+
+        Args:
+            pid: Datapack Id.
+            begin: The span begin index.
+            end: The span end index.
+        Returns:
+            True if the input span overlaps with
+            any existing spans, False otherwise.
+        """
+        if len(self._replaced_annos[pid]) == 0:
+            return False
+        ind: int = (
+            bisect_left(self._replaced_annos[pid], (Span(begin, begin), "")) - 1
+        )
+        if ind < 0:
+            ind += 1
+
+        while ind < len(self._replaced_annos[pid]):
+            span: Span = self._replaced_annos[pid][ind][0]
+            if not (span.begin >= end or span.end <= begin):
+                return True
+            if span.begin > end:
+                break
+            ind += 1
+
+        return False
+
+    def insert_annotations(
+        self, inserted_text: str, data_pack: DataPack, pos: int
+    ) -> bool:
+        r"""
+        This is a utility function to insert a new annotation. After
+        getting the inserted text, it will register the input & output
+        for later batch process of building the new data pack.
+        The insertion at each position can only occur once. If there
+        is already an insertion at current position, it will abort the
+        insertion and return False.
+
+        Args:
+            inserted_text: The text string to insert.
+            `data_pack`: The datapack for insertion.
+            pos: The position(index) of insertion.
+        Returns:
+            A bool value. True if the insertion happened, False otherwise.
+        """
+
+        pid: int = data_pack.pack_id
+        if self._overlap_with_existing(pid, pos, pos):
+            return False
+
+        if pos not in self._inserted_annos_pos_len[pid]:
+            self._replaced_annos[pid].add((Span(pos, pos), inserted_text))
+            self._inserted_annos_pos_len[pid][pos] = len(inserted_text)
+            return True
+        return False
+
+    def delete_annotations(self, input_anno: Annotation) -> bool:
+        r"""
+        This is a utility function to delete an annotation. If the same
+        annotation is tried to be deleted twice, the function will terminate
+        and return False.
+
+        Args:
+            input_anno: The annotation to remove.
+        Returns:
+            A bool value. True if the deletion happened, False otherwise.
+        """
+        pid: int = input_anno.pack.pack_id
+        if self._overlap_with_existing(pid, input_anno.begin, input_anno.end):
+            return False
+
+        self._replaced_annos[pid].add((input_anno.span, ""))
+        self._deleted_annos_id[pid].add(input_anno.tid)
+        return True
+
+    def replace_annotations(
```

Comment:
Do you feel there is a need to include `split_annotation` (like you did before) as another utility?

Reply:
From the work I did in the `word_splitting` op, I feel that the operation of splitting an annotation can be done by the combination of an `insert` and `delete`. So that is why I feel we dont need a separate split utility.